src/main/java/marquez/api/resources/DatasourceResource.java

```diff
@@ -61,6 +64,25 @@ public Response list(
     return Response.ok(DatasourceResponseMapper.toDatasourcesResponse(datasources)).build();
   }
 
+  @GET
+  @ResponseMetered
+  @ExceptionMetered
+  @Timed
+  @Path("/{datasourceId}")
+  @Produces(APPLICATION_JSON)
+  public Response get(@PathParam("datasourceId") final DatasourceId datasourceId)
+      throws MarquezServiceException {
+
+    final Optional<Datasource> datasource;
+    datasource = datasourceService.get(datasourceId.getValue());
+
+    if (datasource.isPresent()) {
+      return Response.ok(DatasourceResponseMapper.map(datasource.get())).build();
+    } else {
+      return Response.status(Response.Status.NOT_FOUND).entity("Datasource not found").build();
```

Comment:
Please define the exception `DatasourceUrnNotFoundException`. As an example, see [`DatasetUrnNotFoundException`](https://github.com/MarquezProject/marquez/blob/feature/dataset-resource-create-get-ops/src/main/java/marquez/api/exceptions/DatasetUrnNotFoundException.java) 

You can then avoid the `if/else` block:

```java
final Datasource datasource =
    datasourceService.get(urn).orElseThrow(() -> new DatasourceUrnNotFoundException(urn));
```

Reply:
Nice :)
I've addressed this in the next review, #394, b/c #387 (this iteration) doesn't have DatasourceUrns.